### ISSUE TITLE
[ENH] LinkedIn Greykite forecasting model interface

### ISSUE THREAD

**Issue by surajitdb** [sktime/sktime#1621]
Please add LinkedIn Greykite time series model as a part of sktime.

**Comment by fkiraly** [comment]
good idea! I'll add the "good first issue" tag, since it seems to be a more or less straightforward case of "fill out the [extension template](https://github.com/alan-turing-institute/sktime/blob/main/extension_templates/forecasting.py)"?

Contributions welcome.

**Comment by Aparna-Sakshi** [comment]
This might be a trivial question, but how does interfacing works? Do we have to import grey kite libraries and fill the extension template or understand the algorithm and then fill the template? @fkiraly @surajitdb

**Comment by aiwalter** [comment]
@Aparna-Sakshi yes you can follow the extension template. We would have to import it as a soft-dependency. How to do this, you can look into how we did it for `fbprophet` or `tbats`.

**Comment by Aparna-Sakshi** [comment]
@aiwalter Thanks a lot for the info, I'll look into it.

**Comment by fkiraly** [comment]
Yes, exactly.

I've turned the observation that there are currently no explicit instructions about soft dependencies into an issue to add this in all extension templates. https://github.com/alan-turing-institute/sktime/issues/1759
You may or may not want to do this on the side, if you are looking into this, @Aparna-Sakshi.

**Comment by baggiponte** [comment]
If you need some inspiration for the code, given the unusual Greykite API, you can draw it from [here](https://github.com/winedarksea/AutoTS/blob/master/autots/models/greykite.py)!

**Comment by fkiraly** [comment]
hm, AutoTS looks a bit like a `pycaret` clone?
But could be a good starting point to what the parameters are etc.

What do you think, @ngupta23?

**Comment by ngupta23** [comment]
AutoTS and Auto_TS came before pycaret time series module. 

AutoTS seems to combine the functionality of sktime and pycaret into a single library (something similar to what Auto_TS also does). 

Also it seems to be using genetic algorithms to find the best models. PyCaret uses random and grid  hyperparameter tuning (adopted from sktime). Having Bayesian hyperparameter optimization as a next step might be good.

Does AutoTS provide the ability to hand craft the individual models if the automated workflow is not desired? I think that kind of flexibility is needed as not everyone is comfortable with a fully automated workflow.

**Comment by fkiraly** [comment]
got it - so the takeaway is, do I understand correctly:
* autoTS predates pycaret, the user facing designs are similar
* but it´s not, in terms of the the internal layers interface: autoTS does not separate between the "composition/specification" layer and the "workflow/deployment" layer. Whereas pycaret builds on sktime as a flexible layer.
* nonetheless, in how autoTS interfaces greykite (the "specific model" layer), we can learn for sktime

**Comment by ngupta23** [comment]
> autoTS predates pycaret-ts (but not pycaret?), the user facing designs are similar

I have not delved much into it, but the focus of `autoTS` seems to be on the "auto" part. Whereas in `pycaret`, the users can customize their workflows for both **manual** (create specific models) and **automated** (pick best from all) approaches. On the other hand, pycaret does not support multivariate but autoTS does. So there might be pros and cons of both libraries.

> nonetheless, in how autoTS interfaces greykite (the "specific model" layer), we can learn for sktime

Yes, I think that is correct.

**Comment by ngupta23** [comment]
@fkiraly It might be interesting to see how **holidays** are handled in `sktime`. Is there an example for how `sktime` does this for `prophet`?

**Comment by fkiraly** [comment]
>It might be interesting to see how holidays are handled in sktime. Is there an example for how sktime does this for prophet?

In-principle, two ways:

* in interfaced "black-box" models that have a "holidays" feature, can be handled only in the context of that model, like `Prophet`
* using transformers that extract holidays, these can be used in a pipeline with any model. Examples would be the `DateTimeFeatures` transformer, or the `HolidayTransformer` #1722 that @mloning wanted to write (but I think he´s abandoned this).

**Comment by Prakruthi12345** [comment]
@fkiraly I can take up this work!

**Comment by fkiraly** [comment]
nice, all yours!

**Comment by vedantag17** [comment]
I was working on implementing the Greykite forecasting model as per the issue.However, while following the example provided in the [Greykite simple forecast documentation](https://linkedin.github.io/greykite/docs/1.0.0/html/gallery/quickstart/0100_simple_forecast.html), I encountered a major issue where the example did not work due to outdated dependencies.
The error I faced was:

```
ValueError:  
All the 3 fits failed.  
It is very likely that your model is misconfigured.  
You can try to debug the error by setting error_score='raise'.  
--------------------------------------------------------------------------------
TypeError: 'function' object is not subscriptable 
```

After debugging, I found that the issue was caused by incompatible dependency versions in Greykite. The solution was to install and update the dependencies to match the latest compatible versions.
 I have created an issue on the greykite repo for the same  [issue #138](https://github.com/linkedin/greykite/issues/138).
